src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -202,35 +318,37 @@ public LanguageExpression GetFullyQualifiedResourceId(ResourceSymbol resourceSym
                 this,
                 context.ResourceScopeData[resourceSymbol],
                 typeReference.FullyQualifiedType,
-                GetResourceNameSegments(resourceSymbol, typeReference));
+                GetResourceNameSegments(resourceSymbol, typeReference),
+                newContext);
         }
 
-        public LanguageExpression GetFullyQualifiedResourceId(ModuleSymbol moduleSymbol)
+        public LanguageExpression GetFullyQualifiedResourceId(ModuleSymbol moduleSymbol, SyntaxBase newContext)
```

Comment:
Is newContext used inside `ScopeHelper.FormatFullyQualifiedResourceId()`? Seemed like it was being passed recursively but I couldn't see other usages.

Reply:
Good call! I was able to remove a few occurrences of it.